bin/restrax

```diff
@@ -405,9 +432,131 @@ class ReStrax(daq_core.DataBases):
         data_docs = sorted(data_docs, key=size, reverse=True)
         return data_docs
 
-    def run_software_veto(self, run_doc: dict):
-        """This is where we can add a software veto for specific runs"""
-        raise NotImplementedError
+    def software_veto_is_on(self, run_doc: dict):
+        """Get if software veto is on from the rundoc"""
+
+        if self.test_software_veto and not self.production:
+
+            return True
+        
+
+    def software_veto_docs(self, data_docs: ty.List[dict]) -> (ty.List[dict], ty.List[dict], ty.List[dict]):
```

Comment:
If you want to simplify this a bit, you could consider using a `set`:
```python
set(['target_1', 'target_2']) - set(['target_2'])
```
```{'target_1'}```

Maybe this will simplify this code a bloc a bit:
```python
def _filter_docs(ts, docs):
    return [d for d in docs if d['type'] in ts]

all_targets = set([d['type'] for d in docs])
no_sv = {t for t in all_targets if f'{t}_sv' not in all_targets}
sv = {t for t in all_targets if t.endswith('_sv')}
originals = all_targets - sv - no_sv
return _filter_docs(docs, sv), _filter_docs(docs, originals), _filter_docs(docs, no_sv)
```

Reply:
Great thanks, implemented!